### PR TITLE
Pass image ID for testing

### PR DESCRIPTION
**Pull request by omron93**
When running `test/run` use image ID instead of IMAGE_NAME. This allows parallel testing of images (for example parallel testing of PRs in Jenkins).

@hhorak @bparees @praiskup Please take a look. I will prepare PRs for other repos later.

### CODE REVIEW

**Comment by hhorak** [comment]
This sounds like a good idea to me.

**Comment by bparees** [comment]
lgtm

File: hack/build.sh [hunk 1]
```diff
@@ -37,7 +37,10 @@ function docker_build_with_version {
   if [[ "${UPDATE_BASE}" == "1" ]]; then
     BUILD_OPTIONS+=" --pull=true"
   fi
-  docker build ${BUILD_OPTIONS} -t ${IMAGE_NAME} -f "${dockerfile}.version" .
+  local build_output=$(mktemp)
+  docker build ${BUILD_OPTIONS} -t ${IMAGE_NAME} -f "${dockerfile}.version" . | tee $build_output
```
Comment (praiskup) [inline hack/build.sh]: This probabyl requires `pipefail` set...  Otherwise we can miss `docker build` failure IMO.

**Comment by praiskup** [comment]
Please consider documenting this in commit message, too.  Otherwise after fixing pipefail issue, it looks like nice improvement.  Thanks!

File: hack/build.sh [hunk 2]
```diff
@@ -72,6 +75,7 @@ for dir in ${dirs}; do
   esac
 
   IMAGE_NAME="${NAMESPACE}${BASE_IMAGE_NAME}-${dir//./}-${OS}"
+  IMAGE_ID=""
```
Comment (omron93) [inline hack/build.sh]: OK, thanks.

File: hack/build.sh (hunk 1 above)
Comment (omron93) [inline hack/build.sh]: > Otherwise we can miss docker build failure IMO.

@praiskup You are right. To catch failure, we have to fix `set -e` also, or not?

**Pull request merged**